packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx

```diff
@@ -0,0 +1,173 @@
+import {Text, vec} from "mafs";
+import * as React from "react";
+
+import {clockwise} from "../../../../util/geometry";
+import {getRayIntersectionCoords as getRangeIntersectionVertex} from "../utils";
+
+import type {CollinearTuple} from "../../../../perseus-types";
+import type {Interval} from "mafs";
+
+interface Props {
+    centerPoint: vec.Vector2;
+    endPoints: [vec.Vector2, vec.Vector2];
+    polygonPoints: readonly vec.Vector2[];
+    active: boolean;
+    range: [Interval, Interval];
+    color?: string;
+}
+
+export const Angle = ({
+    centerPoint,
+    endPoints,
+    range,
+    polygonPoints,
+    color = "var(--movable-line-stroke-color)",
+}: Props) => {
+    const [centerX, centerY] = centerPoint;
+    const areClockwise = clockwise([centerPoint, ...endPoints]);
+    const [[startX, startY], [endX, endY]] = areClockwise
+        ? endPoints
+        : endPoints.reverse();
+
+    const radius = 0.3;
+
+    const a = vec.dist(centerPoint, endPoints[0]);
+    const b = vec.dist(centerPoint, endPoints[1]);
+    const c = vec.dist(endPoints[0], endPoints[1]);
+
+    // Law of cosines
+    const angle = Math.acos((a ** 2 + b ** 2 - c ** 2) / (2 * a * b));
+
+    const startAngle = Math.atan2(startY - centerY, startX - centerX);
+    const endAngle = Math.atan2(endY - centerY, endX - centerX);
+    // const normalizedAngle = angle > 2 * Math.PI ? angle - 2 * Math.PI : angle;
+    const x1 = centerX + radius * Math.cos(startAngle);
+    const y1 = centerY + radius * Math.sin(startAngle);
+    const x2 = centerX + radius * Math.cos(endAngle);
+    const y2 = centerY + radius * Math.sin(endAngle);
```

Comment:
I think there's a simpler way to calculate these `x1,y1,x2,y2` constants that doesn't need trigonometry:

```ts
const y1 = centerY + (startY - centerY) / a * radius;
const x2 = centerX + (endX - centerX) / b * radius;
const x1 = centerX + (startX - centerX) / a * radius;
const y2 = centerY + (endY - centerY) / b * radius;
```

With this approach, you don't need to calculate `startAngle` and `endAngle`.